frontend/ui-pablo/app/defi/utils/bonds/assets.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  BondOffer,
+  BondPrincipalAsset,
+  ConstantProductPool,
+  StableSwapPool,
+} from "@/defi/types";
+import { MockedAsset } from "@/store/assets/assets.types";
+import { matchAssetByPicassoId } from "../assets";
+
+export function getBondPrincipalAsset(
+  bondOffer: BondOffer,
+  supportedAssets: MockedAsset[],
+  lpRewardingPools: Array<StableSwapPool | ConstantProductPool>
+): BondPrincipalAsset {
+  const isLpBasedBond: ConstantProductPool | StableSwapPool | undefined =
```

Comment:
Variable name suggests that this is a boolean, while in reality it is not. I suggest we rename this to `lpBasedBond`.